source/text/fontfaceloader.ts

```diff
@@ -244,77 +208,73 @@ export class FontFaceLoader {
-
-        this._valid = true;
-
-        let text;
-        try {
-            text = await fetchAsync(uri, '', (text) => text);
-        } catch (e) {
-            /* promise rejected */
-            log(LogLevel.Warning, `loading font file '${uri}' failed`);
-            this._valid = false;
-        }
+    static process(fontFace: FontFace, data: string, url: string, headless: boolean = false): FontFace | undefined {
+        const lines = data.split('\n');
 
-        /* promise fulfilled */
-        const lines = text.split('\n');
-
-        const promises = [];
+        const promises = new Array<Promise<void>>();
+        let status = true;
         for (const line of lines) {
-            if (!this._valid) {
-                break;
-            }
-            let values = line.split(' ');
-            const identifier = values[0];
-            values = values.slice(1);
+            let attributes = line.split(' ');
+            const identifier = attributes[0];
+            attributes = attributes.slice(1);
 
-            /* tslint:disable-next-line:switch-default */
             switch (identifier) {
                 case 'info':
-                    this.processInfo(values, fontFace);
+                    status = this.processInfo(attributes, fontFace);
                     break;
+
                 case 'common':
-                    this.processCommon(values, fontFace);
+                    status = this.processCommon(attributes, fontFace);
                     break;
+
                 case 'page':
-                    if (!headless) {
-                        promises.push(this.processPage(values, fontFace, uri));
+                    if (headless) {
+                        break;
+                    }
+                    const promise: Promise<void> | undefined = this.processPage(attributes, fontFace, url);
+                    if (promise !== undefined) {
+                        promises.push(promise);
                     }
                     break;
+
                 case 'char':
-                    this.processChar(values, fontFace);
+                    status = this.processChar(attributes, fontFace);
                     break;
+
                 case 'kerning':
-                    this.processKerning(values, fontFace);
+                    this.processKerning(attributes, fontFace);
+                    break;
+
+                default:
                     break;
             }
+
+            if (status === false) {
+                break;
+            }
         }
 
-        await Promise.all(promises);
+        /* Multiple promises might be invoked (one per page due to async texture2D load). Since this is a non async
+        transform intended to be used in a async fetch, waiting on all promises here. */
+        wait<void>(promises, (reason) => {
+            log(LogLevel.Warning, `processing font face data failed: ${reason}`);
+            status = false;
+        });
```

Comment:
Actually, I find `Promise.all()` preferable, `wait` doesn't seem like a good util function. If you want an explicit log, wrap it in a try-catch block, though an uncaught rejection will be logged on the console anyway, so I don't think it's necessary.

Reply:
yes .. but see above.. wait does not replace Promise.all() it makes it block - it uses this internally as well (or at least it should). That's the whole point.

Reply:
the additional log .. yes we could remove this